### PR TITLE
add redux branch

### PR DESCRIPTION
**Pull request by Aquinasmoriasi**
- Add redux toolkit
- Add actions
- Add reducers
- Add Provider for store

### CODE REVIEW

File: src/components/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (julie-ify) [inline src/components/redux/books/books.js]: - [ ] Refactor both action types to use the ducks pattern, e.g.

  ```
  const ADD_BOOK = 'bookstore/books/ADD_BOOK'
  ```

File: src/components/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const UNDER_CONSTRUCTION = 'Under_Construction';
+const ADD_CATEGORY = 'ADD_CATEGORY';
+
+const categoriesReducer = (state = '', action) => {
```
Comment (julie-ify) [inline src/components/redux/categories/categories.js]: - [ ] Initialize your state to an empty array and set this state to that initial state.

  ```
  const initialState = []
  
  const categoriesReducer = state = initialState, action => {
  // other code
  ```

File: src/components/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const UNDER_CONSTRUCTION = 'Under_Construction';
+const ADD_CATEGORY = 'ADD_CATEGORY';
```
Comment (julie-ify) [inline src/components/redux/categories/categories.js]: - [x] Define your action type here using the ducks pattern

  ```
  const ADD_CATEGORY = 'bookstore/categories/ADD_CATEGORY'
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Aquinasmoriasi,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by vechicin — APPROVED** [review]
# Hi @Aquinasmoriasi ✋, 

Your project is complete! 🎯 🧠 💯 
There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights ⚡ 

- [x] Good job implementing the previous reviewer's requirements.

## Optional suggestions 📂 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Aquinasmoriasi** [comment]
Noted. The changes have been implemented.

**Pull request merged**